uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js

```diff
@@ -187,22 +186,36 @@ define(['controllers/controllers',
 
                     return result;
 
-                }).then(function(result){
+                }).then(function (result) {
 
                     $scope.document.id = result.document.id;
                     $scope.document.case_id = result.document.case_id;
                     $scope.document.file_count = $scope.files.length + uploader.queue.length;
 
+                    $scope.document.open = $scope.openNew;
+
                     AssignmentService.updateTab();
                     $modalInstance.close($scope.document);
                     $scope.$broadcast('ta-spinner-hide');
-                },function(reason){
+                }, function (reason) {
                     CRM.alert(reason, 'Error', 'error');
                     $modalInstance.dismiss();
                     $scope.$broadcast('ta-spinner-hide');
                     return $q.reject();
                 });
-            }
+            };
+
+            $scope.dpOpen = function ($event, name) {
+                $event.preventDefault();
+                $event.stopPropagation();
```

Comment:
It's copy-paste from Tasks Dashboard. I thought the same, but when I removed that it stopped working. I decided not to investigate it right now, as we have limited time.